File: backend/agents/explorer.py
import os
from emergentintegrations.llm.chat import LlmChat, UserMessage
from datetime import datetime, timezone
import uuid
from connectors.github_connector import GitHubConnector
from connectors.jira_connector import JiraConnector

class ExplorerAgent:
    def __init__(self, db, manager):
        self.db = db
        self.manager = manager
        self.llm = LlmChat(
            api_key=os.environ['EMERGENT_LLM_KEY'],
            session_id="explorer",
            system_message="You are an enterprise explorer agent. Analyze existing systems read-only and provide insights, risks, and enhancement proposals. Never modify production systems."
        ).with_model("anthropic", "claude-3-7-sonnet-20250219")
        self.github = GitHubConnector()
        self.jira = JiraConnector()

    async def scan_system(self, system_name: str, repo_url: str = None, jira_project: str = None):
        scan_id = str(uuid.uuid4())
        await self._log(scan_id, "Explorer", f"🔍 Scanning system: {system_name}")
        
        try:
            # Gather system context (mocked connectors)
            context = f"System: {system_name}\n"
            
            if repo_url:
                await self._log(scan_id, "Explorer", "📂 Analyzing repository...")
                repo_data = await self.github.analyze_repo(repo_url)
                context += f"Repository: {repo_data}\n"
            
            if jira_project:
                await self._log(scan_id, "Explorer", "📋 Analyzing Jira project...")
                jira_data = await self.jira.get_project_info(jira_project)
                context += f"Jira: {jira_data}\n"
            
            # AI analysis
            user_message = UserMessage(
                text=f"{context}\n\nProvide: 1) System brief, 2) Risk assessment, 3) Enhancement proposals. Be enterprise-safe."
            )
            
            response = await self.llm.send_message(user_message)
            
            # Create scan record
            scan_doc = {
                "id": scan_id,
                "system_name": system_name,
                "brief": response[:500],
                "risks": ["Data exposure risk", "Legacy dependencies"],
                "proposals": ["API modernization", "Add monitoring"],
                "created_at": datetime.now(timezone.utc).isoformat()
            }
            await self.db.explorer_scans.insert_one(scan_doc)
            
            await self._log(scan_id, "Explorer", "✅ System scan completed")
            
            return scan_doc
        except Exception as e:
            await self._log(scan_id, "Explorer", f"❌ Scan failed: {str(e)}")
            return None

    async def _log(self, scan_id: str, agent_name: str, message: str):
        log_doc = {
            "task_id": scan_id,
            "agent_name": agent_name,
            "message": message,
            "timestamp": datetime.now(timezone.utc).isoformat()
        }
        await self.db.agent_logs.insert_one(log_doc)
        await self.manager.send_log(scan_id, log_doc)

File: backend-go/internal/agents/explorer.go
package agents

import (
	"context"
	"time"

	"github.com/catalyst/backend/internal/config"
	"github.com/catalyst/backend/internal/database"
	"github.com/catalyst/backend/internal/llm"
	"github.com/catalyst/backend/internal/models"
	"github.com/catalyst/backend/internal/websocket"
	"github.com/google/uuid"
)

type ExplorerAgent struct {
	db        *database.Database
	wsManager *websocket.Manager
	llmClient *llm.Client
}

func NewExplorerAgent(db *database.Database, wsManager *websocket.Manager, cfg *config.Config) *ExplorerAgent {
	return &ExplorerAgent{
		db:        db,
		wsManager: wsManager,
		llmClient: llm.NewClient(cfg.LLMKey, cfg.LLMProvider, cfg.LLMModel),
	}
}

func (a *ExplorerAgent) ScanSystem(systemName, repoURL, jiraProject string) error {
	scanID := uuid.New().String()

	a.log(scanID, "Explorer", "🔍 Scanning system: "+systemName)

	// Gather system context (mocked connectors)
	context := "System: " + systemName + "\n"

	if repoURL != "" {
		a.log(scanID, "Explorer", "📂 Analyzing repository...")
		context += "Repository: " + repoURL + " (mocked analysis)\n"
	}

	if jiraProject != "" {
		a.log(scanID, "Explorer", "📋 Analyzing Jira project...")
		context += "Jira: " + jiraProject + " (mocked analysis)\n"
	}

	// AI analysis
	systemPrompt := "You are an enterprise explorer agent. Analyze existing systems read-only and provide insights, risks, and enhancement proposals. Never modify production systems."
	userPrompt := context + "\n\nProvide: 1) System brief, 2) Risk assessment, 3) Enhancement proposals. Be enterprise-safe."

	response, err := a.llmClient.SendMessage(systemPrompt, userPrompt)
	if err != nil {
		a.log(scanID, "Explorer", "❌ Scan failed: "+err.Error())
		return err
	}

	// Create scan record
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	scan := models.ExplorerScan{
		ID:         scanID,
		SystemName: systemName,
		Brief:      response[:min(len(response), 500)],
		Risks:      []string{"Data exposure risk", "Legacy dependencies"},
		Proposals:  []string{"API modernization", "Add monitoring"},
		CreatedAt:  time.Now(),
	}

	a.db.Scans.InsertOne(ctx, scan)

	a.log(scanID, "Explorer", "✅ System scan completed")
	return nil
}

func (a *ExplorerAgent) log(scanID, agentName, message string) {
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	log := models.AgentLog{
		ID:        uuid.New().String(),
		TaskID:    scanID,
		AgentName: agentName,
		Message:   message,
		Timestamp: time.Now(),
	}

	a.db.Logs.InsertOne(ctx, log)
	a.wsManager.SendLog(scanID, map[string]interface{}{
		"task_id":    scanID,
		"agent_name": agentName,
		"message":    message,
		"timestamp":  log.Timestamp.Format(time.RFC3339),
	})
}